appserver/connectors/connectors-internal-api/src/main/java/org/glassfish/connectors/config/validators/ConnectionPoolValidator.java

```diff
@@ -49,49 +50,58 @@
 
 /**
  * Implementation for Connection Pool validation.
- * Following validations are done :
- * - Validation of datasource/driver classnames when resource type is not null 
- * - Max pool size to be always higher than steady pool size
- * - Check if statement wrapping is on when certain features are enabled.
- * 
+ * Following validations are done:
+ * <ul>
+ * <li>Validation of datasource/driver classnames when resource type is not null
+ * <li>Max pool size to be always higher than steady pool size
+ * <li>Check if statement wrapping is on when certain features are enabled.
+ * </ul>
+ *
  * @author Shalini M
  */
-public class ConnectionPoolValidator
-    implements ConstraintValidator<ConnectionPoolConstraint, ResourcePool> {
-    
+public class ConnectionPoolValidator implements ConstraintValidator<ConnectionPoolConstraint, ResourcePool> {
+
     protected ConnectionPoolErrorMessages poolFaults;
-    
+
+
+    @Override
     public void initialize(final ConnectionPoolConstraint constraint) {
         this.poolFaults = constraint.value();
     }
 
-    @Override
-    public boolean isValid(final ResourcePool pool,
-        final ConstraintValidatorContext constraintValidatorContext) {
 
-        if (poolFaults == ConnectionPoolErrorMessages.MAX_STEADY_INVALID) {
-            if (pool instanceof ConnectorConnectionPool) {
-                ConnectorConnectionPool connPool = (ConnectorConnectionPool) pool;
-                String maxPoolSize = connPool.getMaxPoolSize();
-                String steadyPoolSize = connPool.getSteadyPoolSize();
-                if(steadyPoolSize == null) {
-                    steadyPoolSize = Constants.DEFAULT_STEADY_POOL_SIZE;
-                }
-                if (maxPoolSize == null) {
-                    maxPoolSize = Constants.DEFAULT_MAX_POOL_SIZE;
-                }
-                if (Integer.parseInt(maxPoolSize) <
-                        (Integer.parseInt(steadyPoolSize))) {
-                    //max pool size fault
-                    return false;
-                }                
-            }
+    @Override
+    public boolean isValid(final ResourcePool pool, final ConstraintValidatorContext constraintValidatorContext) {
+        if (!ConnectorConnectionPool.class.isInstance(pool)) {
+            return true;
+        }
+        final ConnectorConnectionPool connPool = (ConnectorConnectionPool) pool;
+        if (poolFaults == ConnectionPoolErrorMessages.POOL_SIZE_STEADY) {
+            final int steadyPoolSize = getSteadyPoolSize(connPool);
+            return steadyPoolSize >= 0;
+        } else if (poolFaults == ConnectionPoolErrorMessages.POOL_SIZE_MAX) {
+            final int steadyPoolSize = getSteadyPoolSize(connPool);
+            final int maxPoolSize = getMaxPoolSize(connPool);
+            return maxPoolSize > 0 && maxPoolSize >= steadyPoolSize;
```

Comment:
Is `final` really necessary here?
It's essentially just noise since you're immediately returning.

You could even make it a one liner with `return getSteadyPoolSize(connPool) >= 0;`

Reply:
They are not necessary, but in usual editors (Netbeans/Idea/Eclipse) it is more natural to have all variables visible. It also forces good architecture - you will rather find a good name for private method than recycle variables, which is much harder to understand.
Code with final variables can be very simply optimized by a compiler (not only JIT), it depends on CPU/memory/OS architecture how deep it is effective to go. But today compilers can optimize also non-final variables - and also nanoseconds are not so strong argument, when nearly every code loses hundreds of miliseconds on GC vs. too big variable lifespans, bad synchronizations, resource management ...

Reply:
Fixed.